### PR TITLE
Add option philosophy to docs

### PR DESCRIPTION
**Pull request by lydell**
Note to anyone finding this PR: This is not an attempt to impose _my_ philosophy on Prettier. I've tried hard to write down what I've gathered from the project during its ~1 year of existence.

### CODE REVIEW

File: docs/option-philosophy.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Then there's a bunch of interesting cases.
+
+* `--trailing-comma es5` was added to make it easier to use trailing commas in most environments without having to transpile (trailing function commas were added in ES2017).
+* `--prose-wrap` is important to support all quirky markdown renderers in the wild.
+* `--arrow-parens` was added after [huge demand](https://github.com/prettier/prettier/issues/812). Prettier has to strike a balance between ideal goals and listening to the community.
+* `--jsx-bracket-same-line` was needed for a big company with a huge code base (Facebook), which backed the project when it got started, to be able to [adopt Prettier at all](https://github.com/prettier/prettier/pull/661#issuecomment-295770645).
+
+Finally, perhaps the most interesting of them all is `--bracket-spacing`.
+The truth is that not even [Prettier's creator remembers exactly why it exists](https://github.com/prettier/prettier/issues/715#issuecomment-281096495). It was added super early on without much thought. It now serves as an example of the types of options we should avoid.
```
Comment (suchipi) [inline docs/option-philosophy.md]: The linked comment doesn't actually say that he doesn't remember why it exists, so this is a bit misleading...

Comment (lydell) [reply docs/option-philosophy.md]: That was my interpretation. Do you have a better suggestion?

Comment (suchipi) [reply docs/option-philosophy.md]: After re-reading it, I can see your interpretation now. I think it's fine.

Comment (lydell) [reply docs/option-philosophy.md]: Note that I changed “remembers” to “knows” – is that better?

Comment (suchipi) [reply docs/option-philosophy.md]: Yeah, I think that's closer.

**Comment by jlongster** [comment]
Thanks a lot for this! We meant to write something like this a long time ago.

**Comment by lydell** [comment]
@j-f1 Should we update https://github.com/prettier/prettier/wiki/Common-Replies to point to https://prettier.io/docs/en/option-philosophy.html instead?

**Comment by j-f1** [comment]
@lydel Done :white_check_mark:

**Comment by suchipi** [comment]
Woah, I didn't know that common replies page existed. Nice

**Comment by kachkaev** [comment]
Funny enough, `elm-format` that you are referring here is now a part of one of Prettier community plugins: https://github.com/prettier/prettier/pull/4161  🙌

**Pull request merged**